ocfweb/docs/views/servers.py

```diff
@@ -57,81 +56,74 @@ def has_munin(self):
-        ),
-
-        Host.from_ldap(
-            hostname='pandemic',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'pileup',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='riptide',
-            type='hypervisor',
-        ),
-
-        Host.from_ldap(
-            hostname='corruption',
-            type='server',
-        ),
-
-        Host('blackhole', 'network', 'Managed Cisco Catalyst 2960S-48TS-L Switch.', []),
-
-        Host('logjam', 'printer', '', []),
-        Host('pagefault', 'printer', '', []),
-        Host('papercut', 'printer', '', []),
-        Host.from_ldap('overheat', type='raspi'),
-        Host.from_ldap('tornado', type='nuc'),
-
-        *(Host.from_ldap(desktop, type='desktop')
-          for desktop in sorted(list_desktops())),
-    ]
+    def is_hidden(host):
+        return host['cn'][0][:6] == 'hozer-' or host['cn'][0][:4] == 'dev-'
+
+    def create_hosts(lst):
+        """Accepts a list of raw ldap output, returns a dictionary of host
+        objects indexed by hostname
+        """
+        hosts = {}
+        for h in lst:
+            if not is_hidden(h):
+                description = h.get('description', [''])[0]
+                hostname = h['cn'][0]
+                hosts[hostname] = (Host(hostname, h['type'], description, ()))
+        return hosts
+
+    servers = create_hosts(hosts_by_filter('(type=server)'))
+    desktops = create_hosts(hosts_by_filter('(type=desktop)'))
+    misc = create_hosts(hosts_by_filter('(type=printer)'))
+    hypervisors = {}
+
+    # Add children to hypervisors
+    for h in list(servers.values()):
+        children = []
+        for child_hostname in get_children(h.hostname):
+            child = servers.get(child_hostname, False)
+            if child:
+                servers.pop(child.hostname)
+                children.append(Host(child.hostname, 'vm', child.description, ()))
+        if children:
+            servers.pop(h.hostname)
+            hypervisors[h.hostname] = Host(
+                h.hostname,
+                'hypervisor',
+                h.description,
+                tuple(children),
+            )
+
+    # Handle special cases
+    def change_host_type(hostname, host_type, type_dict):
+        """Pop Host with hostname from servers, change its type, and add to misc
+        """
+        servers.pop(hostname, host_type)
+        type_dict[hostname] = Host.from_ldap(hostname, type=host_type)
+    change_host_type('overheat', 'raspi', misc)
+    change_host_type('tornado', 'nuc', misc)
+    change_host_type('jaws', 'hypervisor', hypervisors)
+    misc['blackhole'] = Host(
+        'blackhole', 'network',
+        'Managed Cisco Catalyst 2960S-48TS-L Switch.', [],
+    )
+
+    return list(hypervisors.values()) + list(servers.values()) \
+        + list(desktops.values()) + list(misc.values())
```

Comment:
NB: it's generally preferred to use parentheses instead of backslashes when possible, i.e.
```python
return (list(hypervisors.values()) + list(servers.values())
        + list(desktops.values()) + list(misc.values()))
```

You can alternatively write this line more compactly using unpacking:
```python
return [*hypervisors.values(), *servers.values(), *desktop.values(), *misc.values()]
```